### DISCUSSION TITLE
Incremental session numbers

### DISCUSSION THREAD

**Discussion by pirilampr** [SilentVoid13/Templater#1177] [Help] [answered]
I am doing an rpg vault (inspired in Nicole van der Hoeven, but i dont have the money to pay for hers), and i am trying to make something similar to the "<% tp.user.getThisGameNum(tp) %>" that she uses, can someone help me whit this? (i am also havin a bad time tring to replicate "<% tp.user.getLastGameTitle(tp) %>" and "<% tp.user.getThisWorld(tp) %>")
Sorry for my english

**Comment by Zachatoo** [comment]
You'll need to describe in more detail what you want. I don't want to have to research how Nicole's vault works and hope that I've even found the correct vault.

1. What are you trying to do?
1. What have you tried?
1. What happens currently?
1. What are you expecting to happen?

**Reply by pirilampr** [reply]
I'm trying to make a number for each session in a specific world. So i have a vault called RPG, and inside that i have a folder for each world. I would like to have an incremental number of sessions inside a world, beggining from 001. But i can't make an templater for that, nether a way for indentifying what is the world that i am currently playing. Thank you so much already for the help!

**Reply by Zachatoo** [reply, marked as answer]
It will depend on your file structure, but assuming your file structure looks like this

```
/
├── Worlds/
│   ├── Cosmere/
│   │   └── Session 001.md
│   │   └── Session 002.md
│   ├── Earth/
│   │   └── Session 001.md
```

You can have a script like this to rename a file based on the world and which session number you're on.

```js
<%*
const folder = tp.file.folder(true);
const world = folder.split("/")[1]; // This will be different depending on how many levels deep in your folder structure your world name is
const sessionNumber = `${app.vault.getAbstractFileByPath(folder).children.length}`.padStart(3, "0");
await tp.file.rename(`Session ${sessionNumber}`);
-%>
```

**Reply by pirilampr** [reply]
thank you man
